src/weapon/w_050.c

```diff
@@ -86,7 +86,7 @@ void EntityWeaponAttack(Entity* self) {
 
 extern u8 D_162000_8017B000[6][8];
 
-s16 func_162000_8017B784(Primitive* prim, s16 x, s16 y) {
+s32 func_162000_8017B784(Primitive* prim, s16 x, s16 y) {
```

Comment:
The new function showed that the signature of this is wrong?

Reply:
yep, s16 on this results in an extra sll v0, v0, 16 in the new function. Has to be s32.

Reply:
I'm not sure. It's expecting an `s32`, but `0004` is expecting an `s16`. `0004` was happy with a return value cast, but `0010` was not, so moving this back to `s32` (I had just changed it with the `0004` PR yesterday) seemed like the most expedient way to move forward.

Reply:
I think it's fine for now, we can continue iterating on it